athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/TimeZoneKey.java

```diff
@@ -0,0 +1,324 @@
+/*-
+ * #%L
+ * Amazon Athena Query Federation SDK
+ * %%
+ * Copyright (C) 2019 - 2020 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connector.lambda.data;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.Set;
+import java.util.TreeMap;
+
+import static java.lang.Character.isDigit;
+import static java.lang.Math.abs;
+import static java.lang.Math.max;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Class for representing timezone to be used in encoding datetime value and timezone value to a single long
+ * a copy from Presto TimeZoneKey
```

Comment:
if this is an exact copy you might need to copy attribution too. Do we really need it to be an exact copy?

Also, we need to be careful about what apache arrow type we are using for this. If a customer actually wants to use TIMESTAMPMILLI that seems to no longer be possible because we are now defining timestampmilli as something else.